packages/driver/src/cross-origin/cypress.ts

```diff
@@ -35,6 +34,41 @@ const createCypress = () => {
     setup(config, env)
   })
 
+  Cypress.specBridgeCommunicator.on('attach:to:window', () => {
+    // It would be ideal to get a window identifier and attach to that window specifically instead of searching all parent windows.
+    const findWindow = () => {
+      for (let index = 0; index < window.parent.frames.length; index++) {
+        const frame = window.parent.frames[index]
+
+        try {
+          const frameHostRegex = new RegExp(`(^|\\.)${ window.location.host.replace(/\./g, '\\.') }$`)
+
+          // Compare host, protocol and test that the window's host ends with the frame's host.
+          // This works because the spec bridge's host is always created without a sub domain.
+          if (window.location.port === frame.location.port
+              && window.location.protocol === frame.location.protocol
+              && frameHostRegex.test(frame.location.host)
+              && window.location.href !== frame.location.href) {
+            return frame
+          }
+        } catch (error) {
+          // Catch DOMException: Blocked a frame from accessing a cross-origin frame.
+          if (error.name !== 'SecurityError') {
+            throw error
+          }
+        }
+      }
+
+      return undefined
+    }
+
+    const autWindow = findWindow()
+
+    if (autWindow) {
+      attachToWindow(autWindow)
+    }
```

Comment:
When the spec bridge is initialized it searches for an AUT of same origin, if we find it we attach otherwise we assume that we'll attach to the AUT onload.

Two basic scenarios are this:

Navigates to a cross origin before creating the spec bridge:

```js
it.('visits cross-origin before', () => {
  cy.visit('localhost') // Top set to localhost
  cy.visit('foobar.com') // AUT navigates to foobar.com
  cy.origin('foobar.com', () => { // Spec bridge created, attaches to AUT
    cy.doStuff() // Hooray we can do stuff in a cross origin
  })
})
```

Navigates to a cross origin before creating the spec bridge:

```js
it.('visits cross-origin after', () => {
  cy.visit('localhost') // Top set to localhost
  cy.origin('foobar.com', () => { // Spec bridge created, no AUT to attach to since it's cross-origin
     cy.visit('foobar.com') // AUT navigates to foobar.com, sends on load message and attaches to the existing spec bridge.
    cy.doStuff() // Hooray we can do stuff in a cross origin
  })
})
```